README.md

```diff
@@ -30,91 +30,127 @@ ParseDate has class methods for date string parsing.
+ParseDate.parse_range('1975 or 1905')                   # last year > first year, raises error
+ParseDate.parse_range('2050')                           # year later than current year + 1, raises error
+ParseDate.parse_range('12345')                          # year later than current year + 1, raises error
+ParseDate.parse_range('random text')                    # can't parse years, raises error
+ParseDate.parse_range(nil)                              # can't parse years, raises error
+
+ParseDate.earliest_year('12/25/00')                     # 2000
+ParseDate.earliest_year('5-1-21')                       # 1921
+ParseDate.earliest_year('1666 B.C.')                    # -1666
+ParseDate.earliest_year('-914')                         # -914
+ParseDate.earliest_year('[c1926]')                      # 1926
+ParseDate.earliest_year('ca. 1558')                     # 1558
+ParseDate.earliest_year('195-')                         # 1950
+ParseDate.earliest_year('199u')                         # 1990
+ParseDate.earliest_year('197?')                         # 1970
+ParseDate.earliest_year('196x')                         # 1960
+ParseDate.earliest_year('18th century CE')              # 1700
+ParseDate.earliest_year('17uu')                         # 1700
+ParseDate.earliest_year('between 1694 and 1799')        # 1694
+ParseDate.earliest_year('between 1 and 5')              # 1
+ParseDate.earliest_year('between 300 and 150 B.C.')     # -300
+ParseDate.earliest_year('1496-1499')                    # 1496
+ParseDate.earliest_year('1750?-1867')                   # 1750
+ParseDate.earliest_year('17--?-18--?')                  # 1700
+ParseDate.earliest_year('1835 or 1836')                 # 1835
+ParseDate.earliest_year('17-- or 18--?')                # 1700
+ParseDate.earliest_year('17th or 18th century?')        # 1600
+ParseDate.earliest_year('ca. 5th–6th century A.D.')     # 400
+ParseDate.earliest_year('ca. 9th–8th century B.C.')     # -999
+ParseDate.earliest_year('ca. 13th–12th century B.C.')   # -1399
+ParseDate.earliest_year('5th century B.C.')             # -599
+
+ParseDate.latest_year('195-')                           # 1959
+ParseDate.latest_year('199u')                           # 1999
+ParseDate.latest_year('197?')                           # 1979
+ParseDate.latest_year('196x')                           # 1969
+ParseDate.latest_year('18th century CE')                # 1799
+ParseDate.latest_year('17uu')                           # 1799
+ParseDate.latest_year('between 1694 and 1799')          # 1799
+ParseDate.latest_year('between 1 and 5')                # 5
+ParseDate.latest_year('between 300 and 150 B.C.')       # -150
+ParseDate.latest_year('1496-1499')                      # 1499
+ParseDate.latest_year('1750?-1867')                     # 1867
+ParseDate.latest_year('17--?-18--?')                    # 1899
+ParseDate.latest_year('1757-58')                        # 1758
+ParseDate.latest_year('1975-05')                        # 1975 (range invalid)
```

Comment:
i know you were saying you just aligned cols on these, but does this raise an error?  if so, worth noting?

Reply:
no, in this case, yyyy-yy, it is sometimes a month, so it makes other assumptions.  2001-05 would be interpreted as 2005 for latest year, but 2019-05 would be 2019.

Reply:
ah, got it, ok.  thanks.